ui/src/services/utils.js

```diff
@@ -82,35 +86,34 @@ export const sortCapacity = createSelector(
       );
 
       const sortedList = list
+        // Filter wrong value (ie: null or incorrect unit)
         .filter(item => k8sCapacityRegex.test(item?.[sortBy]))
         .map(item => {
           /**
-           * item[sortBy].split(k8sCapacityRegex) will return an array of
-           * 4 items, only the 2nd and 3rd are useful.
-           *
-           * We might want to improve the regex to get rid of the useless empty
-           * string at the 1st and 4th place of the array later but right now
-           * just keep the k8s regex
-           *
+           * This regex help us to seperate the capacity into
+           * the size and the unit
            * @example
-           * "1Gi" => ["", "1", "Gi", ""]
-           * "123" => ["", "123", "", ""]
+           * "1Gi" => { 'groups': { size: '1', unit: 'Gi'} }
+           * "123" => { 'groups': { size: '1', unit: undefined } }
            */
-          const [empty, size, unit] = item[sortBy].split(k8sCapacityRegex);
-          const unitBase =
-            sizeUnits.find(sizeUnit => sizeUnit.value === unit)?.base ??
-            sizeUnits[0].value;
+          const sizeRegex = /^(?<size>[1-9][0-9]*)(?<unit>[ikKMGTP]{1,2})?$/;
```

Comment:
minor: could be improved again a little bit by using `^(?<size>[1-9][0-9]*)(?<unit>[kKMGTP]i?)?`

The current one accept `20ii` or `20GG` which is incorrect, the new one is more strict (while not becoming more complex).